akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -658,26 +662,29 @@ object Http extends ExtensionId[HttpExt] with ExtensionIdProvider {
   final case class IncomingConnection(
     localAddress: InetSocketAddress,
     remoteAddress: InetSocketAddress,
-    flow: Flow[HttpResponse, HttpRequest, Unit]) {
+    flow: Flow[HttpResponse, HttpRequest, Promise[Option[Option[InetSocketAddress]]]]) {
 
     /**
      * Handles the connection with the given flow, which is materialized exactly once
      * and the respective materialization result returned.
      */
-    def handleWith[Mat](handler: Flow[HttpRequest, HttpResponse, Mat])(implicit fm: Materializer): Mat =
-      flow.joinMat(handler)(Keep.right).run()
+    def handleWith[Mat](handler: Flow[HttpRequest, HttpResponse, Mat])(implicit fm: Materializer): Mat = {
+      val fillMeInAndUserMat = flow.joinMat(handler)(Keep.both).run()
+      fillMeInAndUserMat._1.success(Some(Some(remoteAddress)))
+      fillMeInAndUserMat._2
+    }
```

Comment:
True, one has to set it somewhere.Where one does it is inverted though: with the attribute, we need to set it _in_ the blueprint, with the materialized value it has to be set at "materialization-site". The second being in more places.